common/persistence/sql/sqlExecutionStore.go

```diff
@@ -859,15 +897,22 @@ func (m *sqlExecutionStore) CompleteCrossClusterTask(
 func (m *sqlExecutionStore) RangeCompleteCrossClusterTask(
 	ctx context.Context,
 	request *p.RangeCompleteCrossClusterTaskRequest,
-) error {
-	if _, err := m.db.DeleteFromCrossClusterTasks(ctx, &sqlplugin.CrossClusterTasksFilter{
+) (*p.RangeCompleteCrossClusterTaskResponse, error) {
+	result, err := m.db.RangeDeleteFromCrossClusterTasks(ctx, &sqlplugin.CrossClusterTasksFilter{
```

Comment:
This is our first time introducing pagination during deletes. How are we going to continue deleting?

Reply:
There are loops in queue processor.